lib/engine/game/g_1822_mx/game.rb

```diff
@@ -229,9 +235,111 @@ def upgrades_to_correct_label?(from, to)
         def stock_round
           G1822MX::Round::Stock.new(self, [
             Engine::Step::DiscardTrain,
-            G1822::Step::BuySellParShares,
+            G1822MX::Step::BuySellParShares,
           ])
         end
+
+        def must_buy_train?(entity)
+          entity.trains.empty? && entity.id != 'NDEM'
+        end
+
+        def sorted_corporations
+          available_corporations = super
+          ndem = corporation_by_id('NDEM')
+          available_corporations << ndem
+          available_corporations
+        end
+
+        def corporation_from_company(company)
+          corporation_by_id(company.id[1..-1])
+        end
+
+        def replace_minor_with_ndem(company)
+          # Remove minor
+          @companies.delete(company)
+          @log << "-- #{company.sym} is removed from the game and replaced with NdeM"
+
+          corporation = corporation_from_company(company)
+          ndem = corporation_by_id('NDEM')
+
+          # Replace token
+          city = hex_by_id(corporation.coordinates).tile.cities[corporation.city]
+          city.remove_reservation!(corporation)
+          city.place_token(ndem, ndem.find_token_by_type)
+          graph.clear
```

Comment:
this is very expensive, do you need to do this?

Reply:
should you just clear the graph for a corporation?

Reply:
I believe that I do need to clear for everyone because the city might now be blocking.  I looked at place_token in the base tokener.rb, and it also does graph.clear.